### ISSUE TITLE
USB and battery charging issues

### ISSUE THREAD

**Issue by Nyryzu** [balopez83/Surface_Pro_3_Hackintosh#12]
Model SP3 i7, EFI release 5.2 with Monterrey

I want to say thank you for the continuous development, I noticed improvement in power management, fans are working less now and it runs generally cooler.

- The USB power issue when it comes to Iphone hotspot is still persistent for me, I tried it again after the new release and I noticed that I don't get "the usb device is drawing too much power" notification anymore but the phone cannot connect still and its not even charging at all.
NB: Android tethering works perfectly so I guess its just an apple thing

- The battery charging is not consistent, when its plugged in, it charges for a long while then it stops and switches to power adapter mode for a few seconds and then it gets back to charging. (If no one else is having this problem then I suspect my charger is faulty)
NB: This was always an issue for me but I forgot to mention it before, its not related to the new release.

- Graphics and , as per Balopez's request, it is working perfectly without any problems at all throughout all kinds of applications...even firefox is working well (I switched to nightly build and its working much better now)

If there are any more thing you want to test out please let me know.

**Comment by balopez83** [comment]
@Nyryzu 

I am glad that these releases are improving your experience though am surprised to hear that some of the bugs are still happening on your computer. 

1) Are you using a hub when connecting the iPhone? If so, have you tried connecting directly to the USB port on the computer itself? macOS requires a download in order to connect an iPhone properly. It won't let you tether or access iTunes and other features until that software downloads. You may want to try plugging in a hub with a wifi adapter running and plug in the iPhone and see if macOS will prompt you to download the software. 

2) Check that you don't have "Optimized battery charging" enabled in settings as shown in the screenshot below. If its enabled you will have weird issues with charging. If that is not enabled, I would recommend trying another charger if you have one as it could be bad. I have not heard anyone else having these issues but its possible others have not noticed yet. 

![Screen Shot 2023-05-18 at 11 25 28 PM](https://github.com/balopez83/Surface_Pro_3_Hackintosh/assets/53441362/8cc6afda-b28f-4649-a8ac-f109f14e4973)

3) Glad to hear this is working well still! I am still having trouble with Firefox but I will try the nightly and see if that improves things. It might just require too much resources for what my SP3 can provide.

**Comment by Nyryzu** [comment]
I never tried a usb hub as I don't own one.
I connect the iphone directly to the surface port,
It recognises it but the connection drops immediately and it infinitely switches back and forth like that. No idea why that happens. I will keep trying a few other things and let you know.

As for the charging, I will do what you said and disable optimised charging and let you know what happens.

Thanks as always ;)

**Comment by Nyryzu** [comment]
Update: I disabled optimized charging and made sure that I wasn't charging the Surface while connecting the phone because at this point I am sure that something is off about my charger, anyway the phone connected properly and I managed to get internet connection from my Iphone.

For everyone reading this: I absolutely recommend sticking to BigSur as it gave me the most stable experience on my SP3 and I got the I7 version. 
It is less resource demanding than Monterey (no idea about Ventura as I am never upgrading to it but I assume it will be even worse on the old SP3), battery life is also much better by a margin of a whole hour. 
These are my 2 cents and I am sure everybody's mileage will vary. 

I just got a new update of Monterey, it may make things better or worse. 
Will post about it after I try it.
